openfisca-web-api/input-output-data.md

```diff
@@ -1,65 +1,40 @@
-# JSON data structures
+# Input and output data
 
-## axes
+## Values by period
 
-A JSON axis is an object structured this way:
-* `count` (integer, >= 1, required): the number of steps to go from min to max
-* `index` (integer, >= 0, default: 0): the index of the person on which to apply the variation of the variable
-* `max` (integer or float, required): the maximum value of the varying variable
-* `min` (integer or float, required): the minimum value of the varying variable
-* `name` (string, one of available [variable names](https://legislation.openfisca.fr/variables), required): the name of the varying variable
-* `period` (a JSON period):
-
-### Parallel axes
-
-TODO
-
-### Perpendicular axes
-
-TODO
-
-## instants
-
-TODO
-
-## parameters
+The value of a variable always exists in a specific period.
 
-TODO
+Example: `{salaire_de_base: {"2015-01": 5000}}`
 
-## periods
+When the period is not given, for example `enfant_a_charge: false`, the global period of the [scenario](#scenarios) is used.
 
-A JSON period can be:
-* a `string` like "2014", "2014-01", "2014-03:2"
-* an `object` like `{start: "2014", unit: "year"}`
+Example: `{salaire_de_base: 5000}` is equivalent to `{salaire_de_base: {<scenario_period>: 5000}}`, `<scenario_period>` being defined in the [scenario](#scenarios).
 
-The Python function used to parse a JSON period is
-[`make_json_or_python_to_period`](https://github.com/openfisca/openfisca-core/blob/master/openfisca_core/periods.py#L1067) (see examples).
-
-## scenarios
+## Scenarios
 
 A JSON scenario is an object structured this way:
 * `axes` (a list of objects, default: null): the axes of the scenario, see [axes](#axes)
 * `input_variables` (an object, mutually exclusive with `test_case`): the input variables, structured this way:
-  * `<variable name (string)>` (the [JSON variable value](#variables-values)): an input variable
-* `period` (a [JSON period](#periods), default: the current year): the period on which the variables of the decomposition will be computed
+  * `<variable name (string)>` (the [value by period](#values-by-period)): an input variable
+* `period` (see [periods and instants](../periodsinstants.md), default: the current year): the period on which the variables of the decomposition will be computed
 * `test_case` (an object, mutually exclusive with `input_variables`): the test case of the scenario, see [test cases](#test-cases)
 
 > Either `test_case` or `input_variables` must be provided, not both.
 
 > `axes` can't be used with `input_variables`, only `test_case`.
 
-## test cases
+## Test cases
 
 A test case describes persons, entities and their associations.
 
 A JSON test case is an object structured this way:
 * `individus` (list of objects): defines persons with their input variables, structured this way:
```

Comment:
I don't understand your comment quite well.

But I agree the description of the structure as a list is not the best way to document it.

> [...] I would prefer to give the general abstract structure [...]

What do you mean more precisely ?